### ISSUE TITLE
Support for AI Search Index resource type

### ISSUE THREAD

**Issue by mrodrig** [hashicorp/terraform-provider-azurerm#25100]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Currently there's no way via Terraform to create an AI Search index after the AI Search instance creation despite there being APIs available for this to use with either `az rest` or directly via `curl`. While it's possible to create an index using a `null_resource` and a `local-exec` provisioner, it's less than ideal. Additionally, trying to support updates to the index based on changes to the index due to the nature of provisioners.

Here's an example curl for creation:

```
curl --location 'https://<search_service_name>.search.windows.net/indexes?api-version=2023-10-01-Preview' \
--header 'api-key: <search_service_key>' \
--header 'content-type: application/json' \
--data '<index_content>'
```

for updates:

```
curl --location --request PUT 'https://<search_service_name>.search.windows.net/indexes/<index_name>?api-version=2023-10-01-Preview' \
--header 'api-key: <search_service_key>' \
--header 'content-type: application/json' \
--data '<index_content>'
```

Looking at the network request from portal.azure.com, it appears the delete request would be:

```
curl --location --request DELETE 'https://<search_service_name>.search.windows.net/indexes/<index_name>?api-version=2020-06-30' \
--header 'api-key:<search_service_key>'
```

### New or Affected Resource(s)/Data Source(s)

azurerm_search_index

### Potential Terraform Configuration

```hcl
resource "azurerm_search_index" "example" {
  name                = "my-index" # index name
  resource_group_name = azurerm_resource_group.example.name
  search_service_name = azurerm_search_service.example.name
  search_service_key  = azurerm_search_service.example.primary_key
  index_content = <<TEMPLATE
{
  "name": "my-index",
  "fields": [
    {
      "name": "id",
      "type": "Edm.String",
      "key": true
    }
  ]
}
TEMPLATE
}
```

**Comment by Joseluismantilla** [comment]
Any updates @rcskosir? 
I'm gonna try with curl commands in the meantime.

**Comment by brandonh-msft** [comment]
> Any updates @rcskosir? I'm gonna try with curl commands in the meantime.

I could not get `curl` commands to work as part of a `local-exec` provisioner; it kept bombing out, likely due to syntax when trying to use the JSON objects.
I ended up having to use the `http` provider's `POST` capabilities in conjunction with `null_resource` and `local-exec` to make it all work fully w/in TF.

I had to do this for indexes, indexers, skillsets, and datasources.

**Comment by brandonh-msft** [comment]
This also seems to be a dupe of a 3-year old issue, #11699

**Comment by rcskosir** [comment]
@mrodrig Thanks for taking the time to submit this issue. This does look like this is a duplicate of #11699, thank you @brandonh-msft for helping spot that. We like to try to keep discussions consolidated, so we’re going to close this new issue in favor of #11699.

**Comment by mrodrig** [comment]
@brandonh-msft I ended up using `az rest` commands in my `null_resource` `local-exec` provisioner since we had the AZ CLI installed in the CI image we're using, so I didn't run into that, but I tested those curl commands and confirmed they were working for our instance when I had opened this. You'll likely find that your `POST` only works on creation and won't work for any updates to the index schema. For that, you're better off using a `PUT` to upsert the index since the AI Search instance can handle some updates to the index, though it's honestly pretty finicky.

@rcskosir I'd argue this issue was requesting a new resource to create the index and #11699 is extremely vague and points towards requesting an argument block within the `azurerm_search_service` to define an index. They both get at a similar idea, but that issue lacks a ton of detail and this one actually has more support reactions, so I'm disappointed with the decision to close this. Hopefully this gets supported eventually, but I guess this just made me lose more hope.
